### ISSUE TITLE
RFE Add File-like objects

### ISSUE THREAD

**Issue by imp** [iredelmeier/filesystem-rs#9]
It looks like right now there is no way to act on files in more than one single operations. If I want to repetitively write to a specific file - I am out of luck. I can only write one using `create_file` or `write_file`. In both cases, the previous content will be lost.
It may be beneficial to have a `File`-like object to allow multiple IO (read or write) to the same file.

**Comment by iredelmeier** [comment]
I'm definitely open to this, and don't remember why I didn't add it initially.

Do you have strong feelings about what methods it makes available? I'm inclined to start with a basic wrapper for `File::open` around `File::write` that implements `io::Read` and `io::Write`, but am open to other suggestions.

**Comment by imp** [comment]
I don't have anything specific in mind - just think that similarity with `std` is a benefit.

**Comment by marad** [comment]
@imp I'm interested in having this feature as well. Are you working on it or found any other solution?

**Comment by imp** [comment]
> @imp I'm interested in having this feature as well. Are you working on it or found any other solution?

No, I am neither working on it nor found any workaround. Still need it though ...

**Comment by petersont** [comment]
Any news on this?  I would love to be able to read/write files a little bit at a time.

**Comment by TheSven73** [comment]
#19 
Hopefully this will go somewhere.
